src/Common/src/Interop/Windows/mincore/Interop.GetConsoleTitle.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.Console_L1, CharSet = CharSet.Unicode, SetLastError = true, EntryPoint = "GetConsoleTitleW")]
+        private static extern int GetConsoleTitle(StringBuilder title, int nSize);
```

Comment:
> Also, I do think StringBuilder better represents buffers of variable lengths

FWIW, I'm not sure how that's relevant in the case of a P/Invoke.  A call to the P/Invoke isn't going to resize the StringBuilder for you to adapt to more data than you've allotted space.  And with the current implementation of StringBuilder, it's a chain of builders rather than a contiguous memory region, so the marshaling layer needs to potentially create a new contiguous region just for the purposes of the call.  Plus, in a case like this where we expect most titles will fit within InitialBufferSizeInChars, if the P/Invoke were defined with a ```char*``` for example, then the initial space could be stack allocated, and subsequent calls could use heap-allocated space, sharing the same P/Invoke signature.

Again, for Title, it's not a big deal.  But we should avoid using StringBuilder for interop in corefx in general.

> Adding [Out] though for explicit annotation.

To be clear, it's more than just being explicit... the default is [In] and [Out], which incurs costs to marshal in both directions, even though in this case marshaling out is all that's required.  Specifying [Out] overrides the default so in marshaling doesn't happen.